src/infrastructure/config.ts

```diff
@@ -4,6 +4,12 @@ import type { RedisConfig } from '@lokalise/node-core'
 const configScope: ConfigScope = new ConfigScope()
 const redisDbValidator = createRangeValidator(0, 15)
 
+type NestedPartial<T> = {
+  [P in keyof T]?: NestedPartial<T[P]>
+}
+
+export type ConfigOverrides = NestedPartial<Config>
```

Comment:
I would move this inside TestContext.ts, and make `ts-deepmerge` a devDependency, we only need this in tests